### PR TITLE
HSEARCH-4182 Improve detection of whether an OutboxEvent entity mapping is required

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4182

See tests in the last commit. Half of them were not passing before this PR.

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/common/spi/SearchIntegrationEnvironment.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		this.classResolver = source.classResolver;
+		this.resourceResolver = source.resourceResolver;
+		this.serviceResolver = source.serviceResolver;
+		beanProvider = source.beanProvider;
+		beanResolver = source.beanResolver;
+	}
+
+	@Override
+	public void close() {
+		beanProvider.close();
```
Comment (fax4ever) [inline engine/src/main/java/org/hibernate/search/engine/common/spi/SearchIntegrationEnvironment.java:88]: Is `beanProvider` supposed to be nullable?

**Review by fax4ever — APPROVED** [review]
LGMT +1 as soos as Ci si fixed

File: engine/src/main/java/org/hibernate/search/engine/common/spi/SearchIntegrationEnvironment.java (hunk 1 above)
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/engine/common/spi/SearchIntegrationEnvironment.java:88]: You are right.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**